File: app/src/main/java/me/aluceps/sandbox/di/ActivityComponent.java
package me.aluceps.sandbox.di;

import dagger.Subcomponent;
import me.aluceps.sandbox.di.scope.ActivityScope;
import me.aluceps.sandbox.view.main.MainActivity;

@ActivityScope
@Subcomponent(modules = ActivityModule.class)
public interface ActivityComponent {

    void inject(MainActivity __);

    FragmentComponent plus(FragmentModule __);
}

File: app/src/main/java/me/aluceps/sandbox/di/ActivityComponent.kt
package me.aluceps.sandbox.di

import dagger.Subcomponent
import me.aluceps.sandbox.di.scope.ActivityScope
import me.aluceps.sandbox.view.main.MainActivity

@ActivityScope
@Subcomponent(modules = [(ActivityModule::class)])
interface ActivityComponent {

    fun plus(module: FragmentModule): FragmentComponent

    fun inject(activity: MainActivity)
}
